go/ct/utils/adapter.go

```diff
@@ -145,8 +144,13 @@ func (c *ctRunContext) Call(kind vm.CallKind, parameter vm.CallParameter) (vm.Ca
 	}, nil
 }
 
-func (c *ctRunContext) SelfDestruct(addr vm.Address, beneficiary vm.Address) bool {
-	panic("not implemented")
+func (c *ctRunContext) SelfDestruct(address vm.Address, beneficiary vm.Address) bool {
+	if c.state.HasSelfDestructed {
+		return false
+	}
+	c.state.HasSelfDestructed = true
+	c.state.SelfDestructedJournal = append(c.state.SelfDestructedJournal, st.NewSelfDestructEntry(address, beneficiary))
```

Comment:
I would move this line to the top of this function, since we want to see all self-destruct calls, even if the account as already self-destructed.